src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -183,9 +183,10 @@ void setupConnection(final Connection connection, final long connectionTimeout)
       connection.setAutoCommit(isAutoCommit);
       connection.setReadOnly(isReadOnly);
 
-      final int defaultLevel = connection.getTransactionIsolation();
-      transactionIsolation = (transactionIsolation < 0 ? defaultLevel : transactionIsolation);
-      if (transactionIsolation != defaultLevel) {
+      if (transactionIsolation == -1) {
+         transactionIsolation = connection.getTransactionIsolation();
+      }
+      else if (transactionIsolation != Connection.TRANSACTION_NONE) {
```

Comment:
but this test fails !?
com.zaxxer.hikari.ConnectionStateTest.testTransactionIsolation(ConnectionStateTest.java:58)